### ISSUE TITLE
go1.7rc6 darwin/amd64 autocompletion panic

### ISSUE THREAD

**Issue by mattetti** [Microsoft/vscode-go#425]
I tried to update to the latest and hopefully final RC of Go 1.7 and I noticed that autocompletion broke. 

![autocompletion-panic](https://cloud.githubusercontent.com/assets/113/17609533/fae9e56c-5fea-11e6-8299-c24ca2762b8c.gif)

I know that the binary output changed in 1.7 and I updated all the go tools vscode reported using. The README says gocode version is used for autocompletion and I wonder if the API changed?
The extension panics but I can't find any specific errors or trace anywhere.

**Comment by mattetti** [comment]
I am using a local build in debug mode and found what I think is the root cause, gocode is hanging when asking for autocompletion:

`gocode -f=json autocomplete "path/to.my/file_test.go" c42`

Doesn't return

**Comment by mattetti** [comment]
This looks suspiciously similar to https://github.com/Microsoft/vscode-go/issues/322 which was fixed. However https://github.com/nsf/gocode/issues/305  seems to indicate gocode doesn't yet support 1.7 With 1.7 out in a few days, this might cause some serious issues for VSCode users.

**Comment by mattetti** [comment]
Good news everyone, I found the root cause. The problem is that I updated Go to 1.7 but Gocode was running its daemon in the background and wasn't aware of the change of version. Gocode uses a server/client architecture and VSCode was trying to communicate with the server but the request was hanging. Restarting gocode solved the problem and everything is great again!

**tl;dr: restart gocode**

If you encounter this issue, in your terminal, type:
```gocode close```

Then restart Code and everything should be working well again.

**Comment by nikhilraog09** [comment]
'gocode' is not recognized as an internal or external command,                                                                                                                                                                                                     
operable program or batch file.

**Comment by brinduc** [comment]
Actually only updating gocode worked for me (restarting didn't)

**Comment by prchoang** [comment]
If use windows 

cd $HOME$/lib/bin

gocode close

example:

cd D:\Development\Go\Lib\bin
gocode close
